blog/_posts/2016-03-17-getting-into-scaladoc-dev.md

```diff
@@ -0,0 +1,136 @@
+
+## How do I generate the docs? ##
+
+Simple:
+
+```bash
+$ git clone git@github.com:scala/scala.git && cd scala
+$ ant docs # "ant docs.lib" if you just want the standard library
+```
+
+## Where to begin ##
+
+The markup of the entity page is defined in
+[Entity.scala](https://github.com/scala/scala/blob/2.12.x/src/scaladoc/scala/tools/nsc/doc/html/page/Entity.scala).
+If you're looking to change the HTML that's where you will want to look first.
+
+If you want to add some static asset, remember that `HtmlFactory` needs to copy
+it to the destination. So be sure to add your new resource to `libResources` in
+[HtmlFactory.scala](https://github.com/scala/scala/blob/2.12.x/src/scaladoc/scala/tools/nsc/doc/html/HtmlFactory.scala).
+
+All the current static assets are located in the
+[lib](https://github.com/scala/scala/tree/2.12.x/src/scaladoc/scala/tools/nsc/doc/html/resource/lib) directory.
+That's where you'd want to put new things (it is also where `index.js` etc live!)
+
+**How can I see my changes?** You have two options, if you changed the markup -
+you'll need to regenerate the docs:
+
+```bash
+$ ant docs.clean && ant docs
+```
+
+When the changes are to scripts or style sheets I use a small `Makefile` in the
+root of the project. This makefile will simply copy the resources to the same location
+as the generated docs:
+
+```make
+all:
+    cp src/scaladoc/scala/tools/nsc/doc/html/resource/lib/{index.css,index.js,template.css,template.js,diagrams.css,diagrams.js} build/scaladoc/library/lib/
+```
+
+Regenerating the docs all the time is a pain - if you just want to generate
+them for a single file, do this:
+
+```bash
+$ ant quick.bin
+$ build/quick/bin/scaladoc path/to/source/File.scala
+```
+
+Lastly, before you open your pull request - test your changes!
+
+```bash
+$ ant test.scaladoc
+```
+
+## Where to really begin ##
+
+There's still a lot of things to do before this can be considered release ready.
+Here's a laundry list of things we need to do:
+
+* General
+    - Async all the things! We should be able to parallelize the generation of
+      the doc pages
+    - CSS cleanup! There are a lot of things not necessary anymore, these could
+      simply be deleted
+* Mobile, the docs look decent on mobile - but there's still a ton to do!
+    - Landscape kills some of our layout
+    - There is no good package overview (list on the right) - ideas welcome!
+* Desktop
+    - Keyboard navigation on entity pages (i.e. between members)
+    - Better keyboard navigation on search results
+    - Minor layout issues
+    - Showing existence of companion `class`/`trait`/`object` (maybe *"You
+      won't believe who* `Companion object Boolean` *is going home with
+      tonight!"*)
+
+The full list of ideas and things to do can be found on
+[scala-dev](https://github.com/scala/scala-dev/issues/84). Comment on the issue or reach out
+to one of us on the [scala/contributors gitter](https://gitter.im/scala/contributors)
```

Comment:
Maybe list peoples' gitter names here so they know to mention the right person? Else anyone interested in contributing may get buried in other messages on scala/contributors?